modules/weko-workflow/weko_workflow/utils.py

```diff
@@ -1525,3 +1528,93 @@ def update_indexes_public_state(item_id):
                 index.public_state = True
     if update_db:
         db.session.commit()
+
+
+def get_account_info(user_id):
+    """Get account's info: email, username.
+
+    :param user_id: User id.
+
+    :return: email, username.
+    """
+    data = get_user_profile_info(user_id)
+    if data:
+        return data.get('subitem_mail_address'), \
+            data.get('subitem_displayname')
+    else:
+        return None, None
+
+
+def get_name_display_hide(list_hide, role):
+    """Get workflow role: displays, hides.
+
+    :param list_hide, role.
+
+    :return: displays, hides.
+    """
+    displays = []
+    hides = []
+    if isinstance(role, list):
+        for tmp in role:
+            if not any(x.id == tmp.id for x in list_hide):
+                displays.append(tmp.name)
+            else:
+                hides.append(tmp.name)
+    return displays, hides
+
+
+def get_displays(list_hide, role):
+    """Get workflow role: displays.
+
+    :param list_hide, role.
+
+    :return: displays.
+    """
+    displays = []
+    if isinstance(role, list):
+        for tmp in role:
+            if not any(x.id == tmp.id for x in list_hide):
+                displays.append(tmp)
+    return displays
+
+
+def save_workflow_role(wf_id, list_hide):
+    """Update workflow role.
+
+    :return:
+    """
+    with db.session.begin_nested():
+        db.session.query(WorkflowRole).filter_by(
+            workflow_id=wf_id).delete()
+        if isinstance(list_hide, list):
+            while list_hide:
+                tmp = list_hide.pop(0)
+                wfrole = dict(
+                    workflow_id=wf_id,
+                    role_id=tmp
+                )
+                db.session.execute(WorkflowRole.__table__.insert(), wfrole)
+    db.session.commit()
+
+
+def get_workflows(workflows):
```

Comment:
This function can be a method of class Workflow.

Thanks,

Reply:
Has been fixed.